[Executorch][Export][1/n] Add exported program and GraphModule as inputs, validation and tests

abhinaykukkadapu opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #16130
* #16129
* #16128
* #16127
* __->__ #16126

Add support for `ExportedProgram` and `GraphModule` as inputs to `ExportSession` with proper validation.

- This allows users to pass already Quantized or exported programs directly to `ExportSession`, skipping the quantization or export stages as needed.
- Includes comprehensive test coverage for all input types.

Differential Revision: [D87576716](https://our.internmc.facebook.com/intern/diff/D87576716/)

abhinaykukkadapu committed:
[Executorch][Export][1/n] Add exported program and GraphModule as inputs, validation and tests

Add support for `ExportedProgram` and `GraphModule` as inputs to `ExportSession` with proper validation.

- This allows users to pass already Quantized or exported programs directly to `ExportSession`, skipping the quantization or export stages as needed.
- Includes comprehensive test coverage for all input types.

Differential Revision: [D87576716](https://our.internmc.facebook.com/intern/diff/D87576716/)

[ghstack-poisoned]

larryliu0820 reviewed (approved):
Review automatically exported from Phabricator review in Meta.

abhinaykukkadapu committed:
Update on "[Executorch][Export][1/n] Add exported program and GraphModule as inputs, validation and tests"


Add support for `ExportedProgram` and `GraphModule` as inputs to `ExportSession` with proper validation.

- This allows users to pass already Quantized or exported programs directly to `ExportSession`, skipping the quantization or export stages as needed.
- Includes comprehensive test coverage for all input types.

Differential Revision: [D87576716](https://our.internmc.facebook.com/intern/diff/D87576716/)

[ghstack-poisoned]

abhinaykukkadapu committed:
Update on "[Executorch][Export][1/n] Add exported program and GraphModule as inputs, validation and tests"


Add support for `ExportedProgram` and `GraphModule` as inputs to `ExportSession` with proper validation.

- This allows users to pass already Quantized or exported programs directly to `ExportSession`, skipping the quantization or export stages as needed.
- Includes comprehensive test coverage for all input types.

Differential Revision: [D87576716](https://our.internmc.facebook.com/intern/diff/D87576716/)

[ghstack-poisoned]
